server/planning/events/utils.py

```diff
@@ -0,0 +1,99 @@
+from datetime import datetime
+from eve.utils import ParsedRequest
+import json
+
+from planning.common import (
+    WORKFLOW_STATE,
+    get_max_recurrent_events,
+)
+from planning.events import EventsAsyncService
+from superdesk.resource_fields import ID_FIELD
+from superdesk.utc import utcnow
+
+
+async def get_series(query, sort, max_results):
+    events_service = EventsAsyncService()
+    page = 1
+
+    while True:
+        # Get the results from mongo
+        req = ParsedRequest()
```

Comment:
This was the old way of doing search, instead you can use `events.find` with the param `use_mongo=True`. For example:

```python
results = await events_service.find(query, sort=sort, max_results=max_results, page=page)
```

If you look at the type signature of that method, you will see `sort` uses `superdesk.core.types.SortParam` (which you can use in the signature of this function)

Reply:
Also, the response from `.find` is our own cursor, so list(results) won't work, you would need to use `docs = await results.to_list_raw())`